src/plugins/sails.js

```diff
@@ -42,7 +42,14 @@ io.sails.reconnection = true;
 // Handles socket updates of game data
 io.socket.on('game', function (evData) {
   switch (evData.verb) {
-    case 'updated':
+    case 'updated': {
+      const currentRoute = router.currentRoute.value;
+      // No-op if the event's gameId doesn't match the url
+      const { gameId: urlGameId } = currentRoute.params;
+      const eventGameId = evData.data?.game?.id ?? evData.data.gameId;
+      if (urlGameId && Number(urlGameId) !== eventGameId) {
+        return;
+      }
```

Comment:
This was the fix. Compare the gameId in the url to the one in the event payload and early-return if they don't match to prevent stale updates